identity-apps-core/components/org.wso2.identity.apps.common/src/test/java/org/wso2/identity/apps/common/listener/AppPortalRoleManagementListenerTest.java

```diff
@@ -0,0 +1,201 @@
+import org.wso2.carbon.identity.role.v2.mgt.core.exception.IdentityRoleManagementException;
+import org.wso2.carbon.identity.role.v2.mgt.core.model.Role;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
+import org.wso2.carbon.user.core.config.RealmConfiguration;
+import org.wso2.identity.apps.common.internal.AppsCommonDataHolder;
+import org.wso2.identity.apps.common.listner.AppPortalRoleManagementListener;
+
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.fail;
+import static org.wso2.carbon.base.CarbonBaseConstants.CARBON_HOME;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.ADMINISTRATOR;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.APPLICATION;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.ORGANIZATION;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.RoleTableColumns.ROLE_NAME;
+import static org.wso2.identity.apps.common.util.AppPortalConstants.CONSOLE_APP;
+
+/**
+ * Test class for AppPortalRoleManagementListener test cases.
+ */
+public class AppPortalRoleManagementListenerTest {
+
+    private AutoCloseable closeable;
+
+    private MockedStatic<PrivilegedCarbonContext> privilegedCarbonContext;
+    private MockedStatic<OrganizationManagementUtil> organizationManagementUtil;
+    private MockedStatic<AppsCommonDataHolder> appsCommonDataHolder;
+
+    @Mock
+    private UserRealm mockUserRealm;
+
+    @Mock
+    private RoleManagementService roleManagementService;
+
+    @Mock
+    private Role role;
+
+    @BeforeMethod
+    public void setUp() {
+
+        setUpCarbonHome();
+        privilegedCarbonContext = mockStatic(PrivilegedCarbonContext.class);
+        organizationManagementUtil = mockStatic(OrganizationManagementUtil.class);
+        appsCommonDataHolder = mockStatic(AppsCommonDataHolder.class);
+        mockCarbonContext(privilegedCarbonContext);
+        closeable = MockitoAnnotations.openMocks(this);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+
+        privilegedCarbonContext.close();
+        organizationManagementUtil.close();
+        appsCommonDataHolder.close();
+        closeable.close();
+    }
+
+    @DataProvider(name = "preUpdateUserListOfRoleDataProvider")
+    public Object[][] preUpdateUserListOfRoleDataProvider() {
+        return new Object[][]{
+            // Test case where deletedUserIDList == null
+            {"roleId", Collections.emptyList(), null, CARBON_HOME, true, false, "adminUserId"},
```

Comment:
Shall we pass some valid tenant domain name like "abc.com" instead of CARBON_HOME?
Make it a constant  for the test class since there are multiple usages

Reply:
Changed the tenant domain to "abc.com" and made all strings class constants.